src/rgw/rgw_lc.cc

```diff
@@ -1241,7 +1241,8 @@ class LCOpAction_NonCurrentExpiration : public LCOpAction {
   int process(lc_op_ctx& oc) override {
     auto& o = oc.o;
     int r = remove_expired_obj(oc.dpp, oc, true,
-                               {rgw::notify::ObjectExpirationNonCurrent});
+                               {rgw::notify::LifecycleExpirationDelete,
```

Comment:
according to documentation: https://docs.ceph.com/en/latest/radosgw/s3-notification-compatibility/
`s3:LifecycleExpiration:Delete` is equivalent to `s3:ObjectLifecycle:Expiration:Current`
while here it is passed together with `s3:ObjectLifecycle:Expiration:NonCurrent`